docker/diffusers-doc-builder/Dockerfile

```diff
@@ -33,7 +33,7 @@ RUN uv pip install --no-cache-dir "git+https://github.com/huggingface/diffusers.
 RUN uv pip install --no-cache-dir \
     accelerate \
     numpy==1.26.4 \
-    hf_transfer \
+    hf_xet \
```

Comment:
Let's keep the changes in `docker` limited to these ones:

https://github.com/huggingface/diffusers/blob/f072c64bf22c6a259d38435b19a8ef3eb67c96a8/.github/workflows/build_docker_images.yml#L73-L77